examples/metering.rs

```diff
@@ -0,0 +1,97 @@
+//! Wasmer will let you easily run Wasm module in a Rust host.
+//!
+//! This example illustrates the basics of using Wasmer through a "Hello World"-like project:
+//!
+//!   1. How to load a Wasm modules as bytes
+//!   2. How to compile the module
+//!   3. How to create an instance of the module
+//!
+//! You can run the example directly by executing in Wasmer root:
+//!
+//! ```shell
+//! cargo run --example metering --release --features "cranelift"
+//! ```
+//!
+//! Ready?
+
+use std::sync::Arc;
+use wasmer::{imports, wat2wasm, Instance, Module, Store};
+use wasmer_compiler_cranelift::Cranelift;
+use wasmer_engine_jit::JIT;
+use wasmer_middlewares::Metering;
+use wasmer::CompilerConfig;
+use wasmer::wasmparser::Operator;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // Let's declare the Wasm module.
+    //
+    // We are using the text representation of the module here but you can also load `.wasm`
+    // files using the `include_bytes!` macro.
+    let wasm_bytes = wat2wasm(
+        br#"
+(module
+  (type $add_one_t (func (param i32) (result i32)))
+  (func $add_one_f (type $add_one_t) (param $value i32) (result i32)
+    local.get $value
+    i32.const 1
+    i32.add)
+  (export "add_one" (func $add_one_f)))
+"#,
+    )?;
+
+    // Create a Store.
+    // Note that we don't need to specify the engine/compiler if we want to use
+    // the default provided by Wasmer.
+    // You can use `Store::default()` for that.
+    let metering = Metering::new(5, |operator| -> u64 {
+        match operator {
+            Operator::LocalGet { .. }
+            | Operator::I32Add { .. }=> 2,
+            _ => {
+                dbg!(operator);
```

Comment:
Is it going to be removed?

Reply:
yes it will, this patch is basically a draft/wip just so I don't loose the code in my stash or somewhere ;)